easygcm-lib/src/main/java/eu/inloop/easygcm/EasyGcm.java

```diff
@@ -0,0 +1,213 @@
+     * {@code SharedPreferences}.
+     *
+     * @param context application's context.
+     * @param regId   registration ID
+     */
+    private static void storeRegistrationId(Context context, String regId) {
+        final int appVersion = GcmUtils.getAppVersion(context);
+        if (sLoggingEnabled) {
+            GcmUtils.Logger.d("Saving regId on app version " + appVersion);
+        }
+        final SharedPreferences.Editor editor = getGcmPreferences(context).edit();
+        editor.putString(PROPERTY_REG_ID, regId);
+        editor.putInt(PROPERTY_APP_VERSION, appVersion);
+        editor.apply();
+    }
+
+
+    /**
+     * Gets the current registration ID for application on GCM service, if there is one.
+     * <p/>
+     * If result is empty, the app needs to register.
+     *
+     * @return registration ID, or empty string if there is no existing
+     * registration ID.
+     */
+    public static String getRegistrationId(Context context) {
+        final SharedPreferences prefs = getGcmPreferences(context);
+        final String registrationId = prefs.getString(PROPERTY_REG_ID, "");
+        if (registrationId.isEmpty()) {
+            if (sLoggingEnabled) {
+                GcmUtils.Logger.d("Registration not found.");
+            }
+            return "";
+        }
+        // Check if app was updated; if so, it must clear the registration ID
+        // since the existing regID is not guaranteed to work with the new
+        // app version.
+        final int registeredVersion = prefs.getInt(PROPERTY_APP_VERSION, Integer.MIN_VALUE);
+        final int currentVersion = GcmUtils.getAppVersion(context);
+        if (registeredVersion != currentVersion) {
+            if (sLoggingEnabled) {
+                GcmUtils.Logger.d("App version changed.");
+            }
+            return "";
+        }
+        return registrationId;
+    }
+
+    /**
+     * Removes the current registration id effectively forcing the app to register again.
+     * @param context application's context.
+     */
+    public static void removeRegistrationId(Context context) {
+        final SharedPreferences prefs = getGcmPreferences(context);
+        prefs.edit()
+                .remove(PROPERTY_REG_ID)
+                .remove(PROPERTY_APP_VERSION)
+                .apply();
+    }
+
+    /**
+     * @return Application's {@code SharedPreferences}.
+     */
+    private static SharedPreferences getGcmPreferences(Context context) {
+        // The registrationId is stored in SharedPreferences by the library.
+        return context.getSharedPreferences(PREFS_EASYGCM, Context.MODE_PRIVATE);
+    }
+
+    @SuppressWarnings("UnusedDeclaration")
+    public void setLoggingEnabled(boolean isEnabled) {
+        sLoggingEnabled = isEnabled;
+    }
+
+    /**
+     * Registers the application defined by a context activity to GCM in case the registration
+     * has not been done already.
```

Comment:
It's important to add information to Javadoc when this method should be called? Application#onCreate? Activity#onCreate?